app/models/organization.rb

```diff
@@ -9,28 +10,36 @@ def initialize(init_params)
     @name = init_params['name']
     @description = init_params['description']
     @imageUrl = init_params['profileImageUrl']
-    @categories = init_params['type']
-    @interests = init_params['categories']
+    @categories = init_params['categories']
+    @interests = init_params['interests']
   end
 
   def self.get_organizations(num, categories, interests)
     orgs = self.get_orgs_from_api()
-
+    csv_text = File.read('lib/interest.csv')
+    interest = CSV.parse(csv_text, headers: true)
     top_orgs = []
     orgs.each do |i|
       org_categories = i['categories']
       org_categories.each do |c|
         categories.each do |quiz_category|
           if c['categoryName'].downcase.include? quiz_category.downcase
+            org_interest = interest.find {|row| row['Organization ID'].to_i == i['organizationId']}
+            if org_interest != nil
+              i['interests'] = org_interest['Organization Interest']
+            else
+              i['interests'] = ''
+            end
             top_orgs.push(Organization.new(i))
           end
         end
       end
     end
 
-    # if interests != nil
-    #   top_orgs = top_orgs.sort_by { |org| -org.count_interest(interests)}
-    # end
+     if interests != nil
+       top_orgs = top_orgs.sort_by { |org| -org.count_interest(interests)}
+     end
+     top_orgs = top_orgs.uniq { |org| org.id}

```

Comment:
Discussed on Slack: The reason it was returning duplicates was due to the substring matching used when matching categories. The decision tree given to us with category names used key words as opposed to the complete Category name as found in the database and so we needed to substring as a workaround. However, substring leads to multiple categories having the same key word in it when only want one match.
Example @Hekaiyo found: 
```{"categoryId"=>2688, "categoryName"=>"Berkeley Public Service Center Designated"}
{"categoryId"=>2694, "categoryName"=>"z ASUC Student Initiated Service Groups"}
{"categoryId"=>5118, "categoryName"=>"Service Student Organizations"}```
one of the categories from the quiz is "Service" and will thus result in 3 orgs being added

Reply:
Removing duplicates is only a bandaid , we have decided to ask customer to provide category names to match database categories.

Reply:
We've decided that it is the LEAD center's responsibility to provide us accurate quiz details, and not ours to guess at which shortened category name corresponds to which actual database category. 

Therefore, we will not be creating some sort of complicated heuristics or hardcoded list of names to solve this issue on this end. The quiz should be correct in the first place.